pkg/controller/elasticsearch/client/autoscaling.go

```diff
@@ -0,0 +1,133 @@
+
+// MachineLearningSettingsGroup is a group of persistent settings.
+type MachineLearningSettingsGroup struct {
+	MaxMemory                   string `json:"xpack.ml.max_ml_node_size"`
+	MaxLazyMLNodes              int32  `json:"xpack.ml.max_lazy_ml_nodes"`
+	UseAutoMachineMemoryPercent bool   `json:"xpack.ml.use_auto_machine_memory_percent"`
+}
+
+func (c *clientV7) CreateAutoscalingPolicy(ctx context.Context, policyName string, autoscalingPolicy esv1.AutoscalingPolicy) error {
+	path := fmt.Sprintf("/_autoscaling/policy/%s", policyName)
+	return c.put(ctx, path, autoscalingPolicy, nil)
+}
+
+func (c *clientV7) DeleteAutoscalingPolicies(ctx context.Context) error {
+	return c.delete(ctx, "/_autoscaling/policy/*", nil, nil)
+}
+
+func (c *clientV7) UpdateMLNodesSettings(ctx context.Context, maxLazyMLNodes int32, maxMemory string) error {
+	return c.put(
+		ctx,
+		"/_cluster/settings",
+		&MachineLearningSettings{
+			&MachineLearningSettingsGroup{
+				MaxLazyMLNodes:              maxLazyMLNodes,
+				MaxMemory:                   maxMemory,
+				UseAutoMachineMemoryPercent: true,
+			}}, nil)
+}
+
+// AutoscalingCapacityResult models autoscaling capacity decisions. It maps each autoscaling policy to its result.
+type AutoscalingCapacityResult struct {
+	Policies map[string]AutoscalingPolicyResult `json:"policies"`
+}
+
+type AutoscalingPolicyResult struct {
+	RequiredCapacity AutoscalingCapacityInfo `json:"required_capacity"`
+	CurrentCapacity  AutoscalingCapacityInfo `json:"current_capacity"`
+	CurrentNodes     []AutoscalingNodeInfo   `json:"current_nodes"`
+}
+
+// AutoscalingCapacityInfo models capacity information as received by the autoscaling Elasticsearch API.
+type AutoscalingCapacityInfo struct {
+	Node  AutoscalingResources `yaml:"node" json:"node,omitempty"`
+	Total AutoscalingResources `yaml:"total" json:"total,omitempty"`
+}
+
+type AutoscalingNodeInfo struct {
+	Name string `json:"name"`
+}
+
+// IsEmpty returns true if all the resource values are empty (no values in the API response).
+// 0 is considered as a value since deciders are allowed to return 0 to fully scale down a tier.
+func (rc AutoscalingCapacityInfo) IsEmpty() bool {
+	return rc.Node.IsEmpty() && rc.Total.IsEmpty()
+}
+
+// AutoscalingCapacity models a capacity value as received by Elasticsearch.
+type AutoscalingCapacity int64
+
+// Value return the int64 value returned by Elasticsearch. It returns 0 if no value has been set by Elasticsearch.
+func (e *AutoscalingCapacity) Value() int64 {
+	if e == nil {
+		return 0
+	}
+	return int64(*e)
+}
+
+// IsEmpty returns true if the value is nil.
+func (e *AutoscalingCapacity) IsEmpty() bool {
+	return e == nil
+}
+
+// IsZero returns true if the value is greater than 0.
+func (e *AutoscalingCapacity) IsZero() bool {
+	return e.Value() == 0
+}
+
+type AutoscalingResources struct {
+	Storage *AutoscalingCapacity `yaml:"storage" json:"storage,omitempty"`
+	Memory  *AutoscalingCapacity `yaml:"memory" json:"memory,omitempty"`
```

Comment:
I suppose it's a quirk of the API that these are just integer values without units?

Reply:
Yes, Elasticsearch does not return units in the result.